Prevent user id field from being dragged into field group

AbdiTolesa opened the pull request:
Fix https://github.com/Strategy11/formidable-pro/issues/5225

![CleanShot 2024-08-01 at 22 36 38](https://github.com/user-attachments/assets/9539845f-a588-4658-9ab9-a6fcbe087858)

Crabcyborg commented on js/formidable_admin.js:
```diff
@@ -788,7 +788,12 @@ function frmAdminBuildJS() {
 			list => {
 				makeDroppable( list );
-				Array.from( list.children ).forEach( child => makeDraggable( child, '.frm-move' ) );
+				Array.from( list.children ).forEach( child => {
```
@AbdiTolesa I don't think you'll want to update the initialization code to fix this.

Your update should be contained to the `allowDrop` function.

Crabcyborg replied:
Going further, you should really only need to update `allowNewFieldDrop`.

AbdiTolesa replied:
@Crabcyborg Yeah I was just updating that already but just saw your comment now.

Crabcyborg reviewed (approved):
Thanks for working on this @AbdiTolesa.

I moved some code around and fixed an issue where you could still drag fields beside a user ID field in a field group that way.

The user ID check makes more sense where the hidden field check is, and the function `groupIncludesBreakOrHidden` needed to also start checking for user id fields since that helper function covers the similar issue with hidden fields.

I think this is good to go now.

🚀

AbdiTolesa commented:
> Thanks for working on this @AbdiTolesa.
> 
> I moved some code around and fixed an issue where you could still drag fields beside a user ID field in a field group that way.
> 
> The user ID check makes more sense where the hidden field check is, and the function `groupIncludesBreakOrHidden` needed to also start checking for user id fields since that helper function covers the similar issue with hidden fields.
> 
> I think this is good to go now.
> 
> 🚀

Thanks @Crabcyborg!
